types.go

```diff
@@ -0,0 +1,55 @@
+package process
+
+import "context"
+
+// Injecter tags a struct as being an expected target of dependency injection.
+type Injecter interface {
+	// Inject populates the fields of the wrapped value in the given meta value.
+	Inject(ctx context.Context, meta *Meta) error
+}
+
+// Initializer wraps behavior that happens on application startup.
+type Initializer interface {
+	// Init is the hook invoked on application startup.
+	Init(ctx context.Context) error
+}
+
+// Process wraps behavior that happens continually through the application
+// lifecycle.
+type Process interface {
+	// Init is the hook invoked on application startup. The process hook is
+	// expected to be long-running. Returning early, prior to the given context
+	// being canceled, is generally an unexpected event.
+	Start(ctx context.Context) error
```

Comment:
The comment for this looks like it needs the name updated. I also think the name should be `Run` instead of `Start` because `Start`, to me, implies something that will begin a process whereas `Run` implies it's the whole thing.

Reply:
Agreed that `Run` is a better name here.